plugins/woocommerce-beta-tester/api/remote-logging/remote-logging.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Internal\Logging\RemoteLogger;
+
+register_woocommerce_admin_test_helper_rest_route(
+	'/remote-logging/status',
+	'get_remote_logging_status',
+	array(
+		'methods' => 'GET',
+	)
+);
+
+register_woocommerce_admin_test_helper_rest_route(
+	'/remote-logging/toggle',
+	'toggle_remote_logging',
+	array(
+		'methods' => 'POST',
+		'args'    => array(
+			'enable' => array(
+				'required'          => true,
+				'type'              => 'boolean',
+				'sanitize_callback' => 'rest_sanitize_boolean',
+			),
+		),
+	)
+);
+
+register_woocommerce_admin_test_helper_rest_route(
+	'/remote-logging/log-event',
+	'log_remote_event',
+	array(
+		'methods' => 'POST',
+	)
+);
+
+register_woocommerce_admin_test_helper_rest_route(
+	'/remote-logging/reset-rate-limit',
+	'reset_php_rate_limit',
+	array(
+		'methods' => 'POST',
+	)
+);
+
+
+/**
+ * Get the remote logging status.
+ *
+ * @return WP_REST_Response The response object.
+ */
+function get_remote_logging_status() {
+	$remote_logger = wc_get_container()->get( RemoteLogger::class );
+
+	return new WP_REST_Response(
+		array(
+			'isEnabled'     => $remote_logger->is_remote_logging_allowed(),
+			'wpEnvironment' => wp_get_environment_type(),
+		),
+		200
+	);
+}
+
+/**
+ * Toggle remote logging on or off.
+ *
+ * @param WP_REST_Request $request The request object.
+ * @return WP_REST_Response The response object.
+ */
+function toggle_remote_logging( $request ) {
+	$enable = $request->get_param( 'enable' );
+
+	if ( $enable ) {
+		update_option( 'woocommerce_feature_remote_logging_enabled', 'yes' );
+		update_option( 'woocommerce_allow_tracking', 'yes' );
+		update_option( 'woocommerce_remote_variant_assignment', 1 );
+	} else {
+		update_option( 'woocommerce_feature_remote_logging_enabled', 'no' );
+	}
+
+	$remote_logger = wc_get_container()->get( RemoteLogger::class );
+	return new WP_REST_Response(
+		array(
+			'isEnabled' => $remote_logger->is_remote_logging_allowed(),
+		),
+		200
+	);
+}
+
+
+/**
+ * Log a remote event for testing purposes.
+ *
+ * @return WP_REST_Response The response object.
+ */
+function log_remote_event() {
+	$remote_logger = wc_get_container()->get( RemoteLogger::class );
+	$result        = $remote_logger->handle(
+		time(),
+		'critical',
+		'Test PHP event from WC Beta Tester',
+		array( 'source' => 'wc-beta-tester' )
+	);
+
+	if ( $result ) {
+		return new WP_REST_Response( array( 'message' => 'Remote event logged successfully.' ), 200 );
+	} else {
+		return new WP_REST_Response( array( 'message' => 'Failed to log remote event.' ), 500 );
+	}
+}
+
+/**
+ * Reset the PHP rate limit.
+ *
+ * @return WP_REST_Response The response object.
+ */
+function reset_php_rate_limit() {
+	global $wpdb;
+	$wpdb->query(
+		"DELETE FROM {$wpdb->prefix}wc_rate_limits"
+	);
+
+	WC_Cache_Helper::invalidate_cache_group( WC_Rate_Limiter::CACHE_GROUP );
+
+	return new WP_REST_Response( array( 'success' => true ), 200 );
+}
```

Comment:
Out of curiosity, why not add all this in a class instead of the global scope?

Reply:
No particular reason. I just follow the same pattern as other existing code in api folder. 😅 

- https://github.com/woocommerce/woocommerce/blob/d6b2429bfe9f8f8eca5e0b4f8630ce935499155e/plugins/woocommerce-beta-tester/api/features/features.php
- https://github.com/woocommerce/woocommerce/blob/d6b2429bfe9f8f8eca5e0b4f8630ce935499155e/plugins/woocommerce-beta-tester/api/live-branches/install.php
- https://github.com/woocommerce/woocommerce/blob/d6b2429bfe9f8f8eca5e0b4f8630ce935499155e/plugins/woocommerce-beta-tester/api/remote-spec-validator/class-wca-test-helper-remote-spec-validator.php

But we do have a few that use classes. 

- https://github.com/woocommerce/woocommerce/blob/d6b2429bfe9f8f8eca5e0b4f8630ce935499155e/plugins/woocommerce-beta-tester/api/remote-inbox-notifications/class-wc-beta-tester-remote-inbox-notifications-helper.php
- https://github.com/woocommerce/woocommerce/blob/d6b2429bfe9f8f8eca5e0b4f8630ce935499155e/plugins/woocommerce-beta-tester/api/remote-inbox-notifications/class-wca-test-helper-remote-inbox-notifications.php

@moon0326 👋  You contribute the most to this beta tester. What do you think about this? Should we start using classes for new files? 

--

I'll merge this PR as is, but we can start using classes for new files or refactor existing ones.